src/app/modules/forms/absence-form/absence-form.component.ts

```diff
@@ -22,36 +40,108 @@ import {MatSelectModule} from "@angular/material/select";
     MatNativeDateModule,
     MatSelectModule,
     MatButtonModule,
-    MatIcon
+    MatIcon,
+    AsyncPipe,
+    NgIf
   ],
   templateUrl: './absence-form.component.html',
   styleUrls: ['./absence-form.component.css']
 })
 export class AbsenceFormComponent {
+  form: FormGroup;
+  isEditMode: boolean;
+  currentDate = new BehaviorSubject(moment());
+  private destroyRef = inject(DestroyRef);
```

Comment:
if you use inject - please use it everywhere and avoid using injection in constructor. Or opposite :)